web/src/settings_components.ts

```diff
@@ -988,22 +1040,70 @@ export function populate_data_for_request(
                     };
                     continue;
                 }
+                data[property_name] = input_value;
+            }
+        }
+    }
+    return data;
+}
 
+export function populate_data_for_group_request(
+    $subsection_elem: JQuery,
+    group: UserGroup,
+): Record<string, string | boolean | number> {
+    const data: Record<string, string | boolean | number> = {};
+    const properties_elements = get_subsection_property_elements($subsection_elem);
+    for (const input_elem of properties_elements) {
+        const $input_elem = $(input_elem);
+        if (check_group_property_changed(input_elem, group)) {
+            const input_value = get_input_element_value(input_elem);
+            if (input_value !== undefined && input_value !== null) {
+                const property_name = extract_property_name($input_elem);
                 if (property_name === "can_mention_group") {
                     data[property_name] = JSON.stringify({
                         new: input_value,
-                        old: group!.can_mention_group,
+                        old: group.can_mention_group,
```

Comment:
This code will only be called for `can_mention_group` and we expect all future group settings to follow the same format, so maybe we can just remove the if condition and call this directly.
And then if we have some other fields which are not group settings using the save discard widget, then we can update this later.

Reply:
Done